### PR TITLE
[FEAT] Bees and Honey!

### PR DESCRIPTION
**Pull request by Amon-S**
# Description

New apiculture system.
This is a continuation of a hold proposal of honey production (https://github.com/sunflower-land/sunflower-land/issues/199), Bees will be unlocked by buying the Bee Hive whose ingredients will be a mixture of crops + animal products + SFL (all recipes are just for testing, this will need to be discussed with the core team for a balanced economy). The use of other animal products (like eggs, manure, meat) is for having a nice progression in-game (the same way one needs to unlock wheat in order to feed animals).

image

Once the hive is unlocked there also be a "flower garden" where one needs to equip a Bee (bought from the hive) in order to pollinate the flower to harvest some delicious honey. Flower pollination will be limited and will have a cooldown for a nice balance. 

image

 

Honey is a delicacy and it will be used to craft various items, the first will be the Queen Bee nft, which will give you a number of Bees every week for free.

image

There will also be a " Bee Box" nft that will increase honey drops from flowers.

image

Last but not least the final item to be crafted is the Sunflower Cake which will give a huge boost on animal production or other in-game perks.

image

This is just the beginning, in the future there could be a Mutant Bee system or different Bee species with a nice breeding system to unlock special bees that will drop different items.

All recipes, supply of bees and prices are just for testing purposes, the core team should be the one deciding these things because they have a better view of the general economy.

a little demonstration video : 
https://streamable.com/bpal5u
 
## Type of change

Please delete options that are not relevant.

 
- [x] New feature (non-breaking change which adds functionality)

## CONTRIBUTORS
 @Z1kou

### CODE REVIEW

File: src/features/game/types/craftables.ts [hunk 1]
```diff
@@ -445,7 +456,37 @@ export const BLACKSMITH_ITEMS: Record<BlacksmithItem, Craftable> = {
       },
     ],
     supply: 100000,
-    disabled: true,
+  },
+
+  "Bee Hive": {
+    name: "Bee Hive",
+    description: "Gather 20% more honey",
```
Comment (brian-man) [inline src/features/game/types/craftables.ts:463]: The PR description says the hive unlocks honey production (meaning > 0).  The description for the item says "20% more".  Which is it?

Comment (Amon-S) [reply src/features/game/types/craftables.ts]: Just noticed it was a mistake that's supposed to be the Bee Box ( that is still not available anywhere to craft yet )

File: src/features/bees/components/Bee.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const { gameService, selectedItem } = useContext(Context);
+  const [
+    {
+      context: { state },
+    },
+  ] = useActor(gameService);
+
+  return (
+    <>
+      {state.inventory["Chicken Coop"] && (
```
Comment (ahh1539) [inline src/features/bees/components/Bee.tsx:20]: checking chicken coop in inventory for display of bee hive?

**Comment by ahh1539** [comment]
Awesome change, I love this! Going to be leaving a thorough review in a few hours

File: src/features/bees/components/Bee.tsx (hunk 2 above)
Comment (Amon-S) [reply src/features/bees/components/Bee.tsx]: Corrected! thanks for pointing it out

**Comment by adamhannigan** [comment]
Amazing work @Amon-S You have gone above and beyond on this implementation ❤️ 

I will need to spend some solid time going through this in depth and working on the API side of things. Just a heads up my current focus is on the migration/token launch so it may take me a few days to give you the thorough feedback you need 🤗

**Comment by xshigo** [comment]
Thank you @adamhannigan for checking this one out!
I hope you and the rest of the team will like the idea.
There is absolutely no need to rush this, I know there are still more important things to look for at the moment before the 9th and also the Barn is not active yet which will need some time for people to understand and use.

**Comment by brian-man** [comment]
Is Honey the first resource that is unlocked by a limited supply NFT?

**Comment by brian-man** [comment]
I'm curious to hear more about how multiple bees works w/r/t bee and flower cooldowns.  A year after this feature is released, what does a "high end" farmer's hive/bee/flower ecosystem look like? How much honey is produced per day on such a farm.

**Comment by Amon-S** [comment]
> Amazing work @Amon-S You have gone above and beyond on this implementation ❤️
> 
> I will need to spend some solid time going through this in-depth and working on the API side of things. Just a heads up my current focus is on the migration/token launch so it may take me a few days to give you the thorough feedback you need 🤗

Hey, Adam thank you for checking this one out!
I understand that there are other things that take priority now and it's absolutely fine (especially token release), the bee system can't be released before the barn system, there will be a need for animal resources in order to unlock the beehive so I first need to see how that works.
I'm getting some nice ideas from the community on how to make this even more interesting and unique in terms of features/gameplay.
Looking forward to working more with you!

**Comment by adamhannigan** [comment]
@Amon-S A few high level bits of feedback.

The connection between the bees and harvesting the flowers is not exactly quite clear from a player perspective. When a user buys bees, they will want to see them buzzing around somewhere. Right now they are hidden.

I suggest we take a similar approach that barn animals have where you need wheat to feed the animals to produce the goods.

In the case of bees consider the following:

1. You buy bees at the barn and they show buzzing around a certain area
2. You must plant & harvest "Flowers" as a normal crop
3. You must feed the flowers to the bees daily
4. You harvest the honey daily from the bees.

The reason I suggest this is so we can remain consistent with how other animal production works. I.e. Plant wheat, feed to animal, collect goods.

If we interest a new style of collecting goods, it means the player needs to learn a new gameplay element and most likely will get confused trying to use it.

**Comment by SteveChickenCX** [comment]
> @Amon-S A few high level bits of feedback.
> 
> The connection between the bees and harvesting the flowers is not exactly quite clear from a player perspective. When a user buys bees, they will want to see them buzzing around somewhere. Right now they are hidden.
> 
> I suggest we take a similar approach that barn animals have where you need wheat to feed the animals to produce the goods.
> 
> In the case of bees consider the following:
> 
> 1. You buy bees at the barn and they show buzzing around a certain area
> 2. You must plant & harvest "Flowers" as a normal crop
> 3. You must feed the flowers to the bees daily
> 4. You harvest the honey daily from the bees.
> 
> The reason I suggest this is so we can remain consistent with how other animal production works. I.e. Plant wheat, feed to animal, collect goods.
> 
> If we interest a new style of collecting goods, it means the player needs to learn a new gameplay element and most likely will get confused trying to use it.

consistency is nice but adding more gameplay elements will vary the experience of the farmers , not everything have to be centered around the: "plant, harvest, collect" cycle.
 

**how about a "small" change that might gamified the game more:**

after buying the bees you could send the bees to different flowers across the map to gather resources.

we can ask the artist to create some flowers designs that will be randomly spawned across the map:


**lets assume we create 5 different flowers:**

1. Red - increase the gain of honey by X%
2. Blue- reduce the flying time after gathering the honey by X%
3. Yellow- give the bees a chance to fly to another flower before returning
4. Golden- small chance X% to gain another bee when returning (RARE FLOWER)
5. White- double the honey gain (RARE FLOWER)

the more flowers the better, we could actually create some fun mechanism here that will be much more gamified and much less subject to botting. 

every 12-48 hours a new set of flowers will be spawned (X flowers at a time) randomly across the map.

the player can send out bees to each the flower that he wants, 
there will be a "flying" distance that the bees need to fly (back and forth) and this will be the cooldown of that bee group.

the more bees you have the more bees you can send out, it can also be possible to send more bees to a certain flower and increase its effectiveness.

**To feed or not to feed** 

we could use the "flying time" as the limiting factor or\ or and add a feeding factor too.

another idea is to make "special plot lands" that are only usable to grow barn food (other than wheat) 

rambled enough, But I hope to see more varied gameplay options :)

**Comment by F1rezd** [comment]
NPC
NPCBIG

**Comment by Amon-S** [comment]
> Is Honey the first resource that is unlocked by a limited supply NFT? That's a type of exclusivity I don't think I've seen in SFF or SFL before.

Technically advanced crops are unlocked with food nfts so it's not the only one

**Comment by Z1kou** [comment]
I read both idea's here and after a bit of giving it thought, I believe we could do a third alternative, @adamhannigan @SteveChickenCX @Amon-S, What about we opt for a system in which you use the UI of the bee hive to lock a set type of flowers with a 24hours cooldown, flowers could have different set of effects like Chicken proposed [ Either Nectar, Or get Wax as a new ressource,that could be used in upcoming recipes, Or increase yield of the crops withing that day(as bees pollinate flowers we can use that to increase the timer of the crops of that day when a certain flower type is locked) ]  I believe this system is innovative, easier to grasp for new players, and easier to implement for the game,what do you guys think !

**Comment by adamhannigan** [comment]
@Amon-S We are so grateful for your effort 🤗 

We will need to put this PR on hold for the time being but will definitely use it as a reference. For the time being the core team need to work on these foundational architecture changes which involve both the API and Smart contracts

**Comment by Amon-S** [comment]
> @Amon-S We are so grateful for your effort 🤗
> 
> We will need to put this PR on hold for the time being but will definitely use it as a reference. For the time being the core team need to work on these foundational architecture changes which involve both the API and Smart contracts

Hey Adam, i was going to delete this pr cause i need to create a new one, i was working on a V2 of the system with your and chicken suggestions, i will show you about all the changes and then make a new PR so you can check the changes when you guys will have time after these days events

**Pull request closed without merging**